### PR TITLE
Show helpful and actionable messages to guide the user to correct extra extension config

### PR DESCRIPTION
**Pull request by chanseokoh**
The user is already forced to set the `implementation` attribute (i.e., `<configuration implementation=...>`) for Maven to be able to parse a complex `<configuration>` structure from the first place. However, this PR further prevents certain cases that can otherwise pass through (e.g., `<configuration>string value`). In such a case, this PR prints out a helpful message:

`[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:2.2.1-SNAPSHOT:dockerBuild (default-cli) on project helloworld: error running extension 'com.example.MyExtension': extension-specific <configuration> for MyExtension is not of type com.example.MyExtensionConfig but java.lang.String; specify the correct type with <pluginExtension><configuration implementation="com.example.MyExtensionConfig"> -> [Help 1]`

And looks like we don't need any extra checks on Gradle. Gradle prints out the usual error message for misconfiguration coming from `Action.execute(someConfigObject)`.
```
* What went wrong:
Execution failed for task ':jibDockerBuild'.
> Could not set unknown property 'foo' for object of type com.example.ExtensionConfig.
```

### CODE REVIEW

**Comment by loosebazooka** [comment]
ah this needs a rebase?

**Comment by chanseokoh** [comment]
It's on top of #2464. #2464 should go in first.

**Comment by chanseokoh** [comment]
Rebased. Ready for review.

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java [hunk 1]
```diff
@@ -458,13 +458,20 @@ public JibContainerBuilder runPluginExtensions(
       ContainerBuildPlan buildPlan)
       throws JibPluginExtensionException {
     T extraConfig = null;
-    if (extraConfigType.isPresent() && config.getExtraConfiguration().isPresent()) {
-      try {
+    if (config.getExtraConfiguration().isPresent()) {
+      if (!extraConfigType.isPresent()) {
+        log(
```
Comment (loosebazooka) [inline jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java:463]: maybe this should be a hard failure?

Comment (loosebazooka) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java:463]: for example if you try to specify `jib.to.potato` I think you get a failure?

Comment (chanseokoh) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java]: Makes sense. Done.

**Pull request merged**